swarm_dial.go

```diff
@@ -56,6 +59,11 @@ var (
 	ErrGaterDisallowedConnection = errors.New("gater disallows connection to peer")
 )
 
+var fdConsumingTptProtos = map[int]struct{}{
+	ma.P_WS:  struct{}{},
+	ma.P_TCP: struct{}{},
+}
```

Comment:
I'd strongly prefer to just check if the address contains `tcp` and/or `unix` (i.e., a fd consuming transports) before any proxy addresses. Otherwise, we're going to miss transports (e.g., WSS).

Reply:
I think I grok what you mean here and in https://github.com/libp2p/go-libp2p-swarm/pull/212#discussion_r427025170. I've made the changes.